### PR TITLE
Add support for mariadbdatabase with webhook

### PR DESCRIPTION
**Pull request by TasdidurRahman**

### CODE REVIEW

**Review by spectro30 — DISMISSED** [review]
LGTM

**Review by kamolhasan — CHANGES_REQUESTED** [review]

File: apis/schema/v1alpha1/mariadbdatabase_helpers.go [hunk 1]
```diff
@@ -19,7 +19,13 @@ package v1alpha1
 import (
 	"kubedb.dev/apimachinery/crds"
 
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"kmodules.xyz/client-go/apiextensions"
+	"kmodules.xyz/client-go/meta"
+)
+
+const (
+	MariaDBSuffix string = "mariadb"
```
[the remark this answers is not in the document]
Comment (TasdidurRahman) [reply apis/schema/v1alpha1/mariadbdatabase_helpers.go]: done

**Review by spectro30 — APPROVED** [review]

File: apis/schema/v1alpha1/mariadbdatabase_webhook.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+	if name == DatabaseNameAdmin {
+		return field.Invalid(path, in.Name, `cannot use "admin" as the database name`)
+	}
+	if name == DatabaseNameConfig {
+		return field.Invalid(path, in.Name, `cannot use "config" as the database name`)
+	}
+	return nil
+}
+
+func (in *MariaDBDatabase) validateMariaDBDatabaseNamespace() *field.Error {
```
Comment (kamolhasan) [inline apis/schema/v1alpha1/mariadbdatabase_webhook.go:194]: Remove ns validation

File: apis/schema/v1alpha1/mariadbdatabase_webhook.go [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+	if ns == "kubevault" {
+		return field.Invalid(path, in.Name, `cannot use namespace "kubevault" to create schema manager`)
+	}
+	if ns == "local-path-storage" {
+		return field.Invalid(path, in.Name, `cannot use namespace "local-path-storage" to create schema manager`)
+	}
+	return nil
+}
+
+func (in *MariaDBDatabase) validateMariaDBDatabaseName() *field.Error {
```
Comment (kamolhasan) [inline apis/schema/v1alpha1/mariadbdatabase_webhook.go:218]: Remove Lenght validation, use the default validation

**Review by kamolhasan — CHANGES_REQUESTED** [review]

File: apis/schema/v1alpha1/mariadbdatabase_webhook.go (hunk 2 above)
Comment (TasdidurRahman) [reply apis/schema/v1alpha1/mariadbdatabase_webhook.go]: removed

File: apis/schema/v1alpha1/mariadbdatabase_webhook.go (hunk 3 above)
Comment (TasdidurRahman) [reply apis/schema/v1alpha1/mariadbdatabase_webhook.go]: removed

**Review by kamolhasan — APPROVED** [review]
LGTM!

**Review by spectro30 — APPROVED** [review]

**Pull request merged**